lib/matplotlib/patches.py

```diff
@@ -33,8 +34,20 @@ class Patch(artist.Artist):
     are *None*, they default to their rc params setting.
     """
     zorder = 1
-    validCap = mlines.Line2D.validCap
-    validJoin = mlines.Line2D.validJoin
+
+    @_api.deprecated("3.4")
+    @_api._classproperty
+    def validCap(cls):
+        with warnings.catch_warnings():
+            cbook._suppress_matplotlib_deprecation_warning()
```

Comment:
```suggestion
        with cbook._suppress_matplotlib_deprecation_warning():
```

Reply:
herp derp, obviously should use the context manager as a context manager, thanks!